libs/novui/src/components/toggle/Toggle.tsx

```diff
@@ -0,0 +1,34 @@
+import { PolymorphicRef } from '../../types/props-helpers';
+import { ChangeEvent, forwardRef } from 'react';
+import { Switch } from '@mantine/core';
+import { css, cx } from '../../../styled-system/css';
+import { toggle } from '../../../styled-system/recipes';
+import { splitCssProps } from '../../../styled-system/jsx';
+import { CoreProps, LocalizedMessage } from '../../types';
+
+export type ToggleProps = CoreProps & {
+  label?: LocalizedMessage;
+  checked?: boolean;
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+  disabled?: boolean;
+};
+
+export const Toggle = forwardRef((props: ToggleProps, ref?: PolymorphicRef<'input'>) => {
+  const [variantProps, toggleProps] = toggle.splitVariantProps(props);
+  const [cssProps, localProps] = splitCssProps(toggleProps);
+  const { onChange, className, checked, disabled, label, required, readOnly } = localProps;
+  const classNames = toggle(variantProps);
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** Nice job getting this setup!‏